dlib/python/numpy.h

```diff
@@ -30,26 +32,22 @@ void validate_numpy_array_type (
 
 // ----------------------------------------------------------------------------------------
 
-template <typename T, int dims>
-void get_numpy_ndarray_parts (
+template <int dims>
+void get_numpy_ndarray_shape (
```

Comment:
Why are there two versions of get_numpy_ndarray_shape().  It seems like you should only need the const version.